test/alternator/test_gsi.py

```diff
@@ -163,9 +163,25 @@ def test_gsi_strong_consistency(test_table_gsi_1):
 
 # Test that setting an indexed string column to an empty string is illegal,
 # since keys cannot contain empty strings
-def test_gsi_empty_value(test_table_gsi_2):
+def test_gsi_empty_value(test_table_gsi_3):
     with pytest.raises(ClientError, match='ValidationException.*empty'):
-        test_table_gsi_2.put_item(Item={'p': random_string(), 'x': ''})
+        test_table_gsi_3.put_item(Item={'p': random_string(), 'a': ''})
+    with pytest.raises(ClientError, match='ValidationException.*empty'):
+        test_table_gsi_3.put_item(Item={'p': random_string(), 'b': ''})
+
+# Dynamodb supports special way of setting NULL value.
+# It's different than non existing value (see also test_update_expected_1_not_null).
```

Comment:
This comment is a bit misleading, because test_update_expected_1_not_null is not actually about the "NULL" value (as its comments explain).

Reply:
I meant to add this ref as an expansion of the topic, will remove